netbox/extras/models/scripts.py

```diff
@@ -137,9 +137,13 @@ def sync_classes(self):
         Syncs the file-based module to the database, adding and removing individual Script objects
         in the database as needed.
         """
-        db_classes = {
-            script.name: script for script in self.scripts.all()
-        }
+        if self.id:
+            db_classes = {
+                script.name: script for script in self.scripts.all()
+            }
+        else:
+            db_classes = {}
+
```

Comment:
Is this change necessary? If `self.id` is null, `self.scripts.all()` should always return an empty queryset IIRC, resulting in an empty dict.

Reply:
Yes, it's because you can't ref an M2M before you save, if this isn't here and the ScriptModule hasn't been saved (no id) then you will get an 'ValueError: 'ScriptModule' instance needs to have a primary key value before this relationship can be used.'